google/web-stories-wp
# Improve embed block markup and AMP for WP compatibility

## Context



We got a user report about an incompatibility with the AMP for WP plugin, where our embed block is removed completely and causing AMP validation errors (see screenshots below).

Unfortunately the AMP for WP plugin is really inferior to the official AMP plugin and does not automatically add missing attributes to ensure validity, nor does it support extracting stylesheets  like the official AMP (Page Experience) plugin does. 😞 

## Summary



This PR improves embed block markup and improves compatibility with the AMP for WP plugin.

* Adds some required element attributes like `tabindex` which the official AMP plugin would normally add for us automatically.
* Hooks into `amp_post_template_css` provided by AMP for WP to load the necessary stylesheets.

## Relevant Technical Choices



* Adds embeds styles as inline stylesheet for the AMP for WP plugin
* Improve embeds markup to include proper `<a>` links for improved accessibility and SEO
* Fixes accessibility issue with the lightbox close button

## To-do



N/A

## User-facing changes



Before:

![Screenshot 2021-10-08 at 17 25 41](https://user-images.githubusercontent.com/841956/136583118-0e3bc51e-558f-4a56-b265-7a26e2ede612.png)

After:

![Screenshot 2021-10-08 at 17 25 30](https://user-images.githubusercontent.com/841956/136583156-66781c24-353f-4f53-9bfb-310fc7eb3efd.png)


## Testing Instructions






This PR can be tested by following these steps:

1. Activate the AMP for WP plugin (not the official AMP plugin)
2. Publish a new blog post and insert a Latest Stories Web Stories block
3. View the AMP version of the blog post on the site and verify that the block works as intended


## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #9266
Fixes #9314
